invokeai/frontend/web/src/features/controlNet/components/parameters/ParamControlNetWeight.tsx

```diff
@@ -1,32 +1,18 @@
-import { createSelector } from '@reduxjs/toolkit';
-import { stateSelector } from 'app/store/store';
-import { useAppDispatch, useAppSelector } from 'app/store/storeHooks';
-import { defaultSelectorOptions } from 'app/store/util/defaultMemoizeOptions';
+import { useAppDispatch } from 'app/store/storeHooks';
 import IAISlider from 'common/components/IAISlider';
-import { controlNetWeightChanged } from 'features/controlNet/store/controlNetSlice';
-import { memo, useCallback, useMemo } from 'react';
+import {
+  ControlNetConfig,
+  controlNetWeightChanged,
+} from 'features/controlNet/store/controlNetSlice';
+import { memo, useCallback } from 'react';
 
 type ParamControlNetWeightProps = {
-  controlNetId: string;
+  controlNet: ControlNetConfig;
 };
 
 const ParamControlNetWeight = (props: ParamControlNetWeightProps) => {
-  const { controlNetId } = props;
+  const { weight, isEnabled, controlNetId } = props.controlNet;
```

Comment:
All of these controlnet config changes are unrelated to multiselect, right?

Reply:
Yes - I kept on running into bugs related to indexed array access that gave me `undefined` when I expected an `ImageDTO`, due to the array length. Typescript does not mark indexed access as possibly undefined by default.

I enabled `noUncheckedIndexedAccess` in `tsconfig` to make Typescript be stricter with this, which showed me all the places where this class of but could pop up. 

Unfortunately, it was a lot of places unrelated to the gallery, and I fixed all occurrences.